scripts/screenshot.js

```diff
@@ -0,0 +1,117 @@
+const puppeteer = require("puppeteer");
+const cliProgress = require("cli-progress");
+const config = require("./screenshot.config.json");
+
+// Remove these selectors in all scenarios.
+const globalRemoveSelectors = ["#__bs_notify__", ".visual-regression-remove"];
+
+// Hide these selectors in all scenarios.
+const globalHideSelectors = [
+  "iframe[src*='youtube']",
+  "img[src*='picsum']",
+  ".visual-regression-hide",
+];
+
+const removeSelectors = async (page) => {
+  return Promise.all(
+    globalRemoveSelectors.map(async (selector) => {
+      await page.evaluate((sel) => {
+        document.querySelectorAll(sel).forEach((s) => {
+          s.style.cssText = "display: none !important;";
+          s.classList.add("__86d");
+        });
+      }, selector);
+    })
+  );
+};
+
+const hideSelectors = async (page) => {
+  return Promise.all(
+    globalHideSelectors.map(async (selector) => {
+      await page.evaluate((sel) => {
+        document.querySelectorAll(sel).forEach((s) => {
+          s.style.visibility = "hidden";
+        });
+      }, selector);
+    })
+  );
+};
+
+const screenshot = async (browser, viewportName, pageName) => {
+  const page = await browser.newPage();
+  const filename = `${pageName}-${viewportName}.png`;
+  const viewport = config.viewports[viewportName];
+  const uri = config.uris[pageName];
+  await page.setViewport(viewport);
+  await page.goto(uri, { waitUntil: "load", timeout: 0 });
+  // Give it 2 seconds for images etc to load and animations to fire.
+  await new Promise((res) => {
+    setTimeout(() => res(), 2000);
+  });
+
+  if (globalHideSelectors.length > 0) await hideSelectors(page);
+  if (globalRemoveSelectors.length > 0) await removeSelectors(page);
+
+  await page.screenshot({
+    path: `screenshots/${filename}`,
```

Comment:
Consider adding fullPage: true, to these parameters if you will have components/screenshots that extend beyond the visible viewport.

Reply:
Added